dataloaders/objectron_main.py

```diff
@@ -0,0 +1,184 @@
+        self.mode = mode
+        if mode == 'train':
+            ann_path = Path(root_folder).resolve() /  'annotations/objectron_train.json'
+            with open(ann_path, 'r') as f:
+                self.ann = json.load(f)
+        elif mode in ['val', 'test']:
+            ann_path = Path(root_folder).resolve() /  'annotations/objectron_test.json'
+            with open(ann_path, 'r') as f:
+                self.ann = json.load(f)
+        else:
+            raise RuntimeError("Unknown dataset mode")
+
+    def __len__(self):
+        return len(self.ann['annotations'])
+
+    def __getitem__(self, indx):
+        # get path to image from annotations
+        raw_keypoints = self.ann['annotations'][indx]['keypoints']
+        img_id = self.ann['annotations'][indx]['image_id']
+        category = self.ann['annotations'][indx]['category_id']
+        # get raw key points for bb from annotations
+        img_path = self.root_folder + '/' + (self.ann['images'][img_id]['file_name'])
+        # read image
+        image = cv.imread(img_path, flags=1)
+        assert image is not None
+        # The keypoints are [x, y] where `x` and `y` are unnormalized
+        # transform raw key points to this representation
+        unnormalized_keypoints = np.array(raw_keypoints).reshape(9, 2)
+        # "print" image after crop with keypoints if needed
+        if self.debug_mode:
+            imgh = image.copy()
+            b = np.zeros((9,3))
+            b[:,:2] = self.normalize(image, unnormalized_keypoints)
+            graphics.draw_annotation_on_image(imgh, b , [9])
+            cv.imwrite('image_before_pipeline.jpg', imgh)
+        # given unnormalized keypoints crop object on image
+        cropped_keypoints, cropped_img = self.crop(image, unnormalized_keypoints)
+
+        # convert colors from BGR to RGB
+        cropped_img = cv.cvtColor(cropped_img, cv.COLOR_BGR2RGB)
+        # do augmentations with keypoints
+        if self.transform:
+            transformed = self.transform(image=cropped_img, keypoints=cropped_keypoints)
+            assert np.array(transformed['keypoints']).shape == np.random.rand(9,2).shape
+
+            transformed_image = transformed['image']
+            transformed_bbox = transformed['keypoints']
+        else:
+            transformed_image, transformed_bbox = image, cropped_keypoints
+        # given croped image and unnormalized key points: normilize it for cropped image
+        transformed_bbox = self.normalize(transformed_image, transformed_bbox)
+        # "print" image after crop with keypoints if needed
+        if self.debug_mode:
+            imgh = transformed_image.copy()
+            b = np.zeros((9,3))
+            b[:,:2] = transformed_bbox
+            graphics.draw_annotation_on_image(imgh, b , [9])
+            cv.imwrite('after_preprocessing.jpg', cv.cvtColor(imgh, cv.COLOR_RGB2BGR))
+
+        transformed_image = np.transpose(transformed_image, (2,0,1)).astype(np.float32)
```

Comment:
I'd propose to move all "to tensor" and related stuff to a albumentation-style transformation. Debugging mode could be also implemented in this way.